File: src/my_project/easy_problems/from101to150/averageSalaryMinMax.py
from typing import List 

class Solution:
    def average(self, salary: List[int]) -> float:


        max_number: int = max(salary)
        min_number: int = min(salary)

        salary.sort()
        total_sum: int = 0

        for i in range(1, len(salary)-1,1):
            total_sum += salary[i]

        return total_sum/(len(salary)-2)


solution = Solution()

print(solution.average(salary = [4000,3000,1000,2000]))

print(solution.average(salary = [6000,5000,4000,3000,2000,1000]))

File: src/my_project_typescript/easy_problems_typescript/from1to50/averageSalaryMinMax.ts
function average(salary: number[]): number {

    let maxNumber: number = Math.max.apply(null, salary);
    let minNumber: number = Math.min.apply(null, salary);


    salary = salary.sort((a,b) => a<=b?-1:1)

    let totalSum: number = 0;

    for (let i = 1; i < salary.length-1; i++){
        totalSum += salary[i]
    }
    
    return totalSum/(salary.length-2)
};


console.log(average([6000,5000,4000,3000,2000,1000]))